tools/protoxform/migrate.py

```diff
@@ -47,7 +56,19 @@ def UpgradeType(match):
         type_desc = self._typedb.types[api_v2_type_name]
       repl_type = type_desc.next_version_type_name[len(
           'envoy.'):] if type_desc.next_version_type_name else normalized_type_name
-      return '<envoy_api_%s_%s%s>' % (ref_type, repl_type, '.' + residual if residual else '')
+      # TODO(htuch): this should really either go through the type database or
+      # via the descriptor pool and annotations, but there are only two of these
+      # we need for the initial v2 -> v3 docs cut, so hard coding for now.
```

Comment:
Can you file a tracking issue for cleanup on this ASAP?

Reply:
Done, see https://github.com/envoyproxy/envoy/issues/9734.